### PR TITLE
``consider-using-with`` now triggers for ``pathlib.Path.open``

### LINKED ISSUE #7964: `consider-using-with` (`R1732`) doesn't trigger for `Path.open()`
### Current problem

```py
file = open('foo')          # consider-using-with
file = Path('foo').open()   # no error
path = Path('foo')
file = path.open()          # no error
```

### Desired solution

`consider-using-with` should trigger on any `Path.open` call too.
I can try to implement this if it's not too hard.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by GideonBear**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |

## Description

Closes #7964

### CODE REVIEW

**Comment by GideonBear** [comment]
In Primer / Run / Run / 3.11:
> Error: Unhandled error: HttpError: API rate limit exceeded for 40.122.244.99

File: tests/functional/c/consider/consider_using_with.py [hunk 1]
```diff
@@ -2,13 +2,24 @@
...
 from concurrent.futures import ThreadPoolExecutor, ProcessPoolExecutor
+from pathlib import Path
+
+
+def test_pathlib_open():
+    _ = pathlib.Path("foo").open(encoding="utf8")  # [consider-using-with]
+    _ = Path("foo").open(encoding="utf8")  # [consider-using-with]
+    path = Path("foo")
+    _ = path.open(encoding="utf8")  # [consider-using-with]
+    with Path("foo").open(encoding="utf8") as file:  # must not trigger
```
Comment (clavedeluna) [inline tests/functional/c/consider/consider_using_with.py:21]: ` # must not trigger` comment not really necessary

**Review by clavedeluna — APPROVED** [review]

File: tests/functional/c/consider/consider_using_with.py (hunk 1 above)
Comment (GideonBear) [reply tests/functional/c/consider/consider_using_with.py:21]: I saw it in other tests, so I thought I should add it too. Should I remove it?

Comment (clavedeluna) [reply tests/functional/c/consider/consider_using_with.py:21]: I vote yes but it's also not a big deal at all.

**Review by Pierre-Sassoulas — APPROVED** [review]
LGTM,thank you @GideonBear

File: tests/functional/c/consider/consider_using_with.py (hunk 1 above)
Comment (Pierre-Sassoulas) [reply tests/functional/c/consider/consider_using_with.py:21]: Yeah it's redundant, probably better to keep it for where it's ambiguous if it should raise or if it's an expected false negative but also let's not rerun a CI for that :)

**Pull request merged**